file_loader.py

```diff
@@ -0,0 +1,328 @@
+
+        file_names.append(nii_file)
+        dimensions_list.append(dimensions)
+        pixel_sizes.append(pixel_size)
+
+    # Create a DataFrame
+    df = pd.DataFrame({
+        'File Name': file_names,
+        'Dimensions': dimensions_list,
+        'Pixel Size': pixel_sizes
+    })
+
+    return df
+
+
+def copy_files_to_bids_folder(contrast, fname, output_folder, participant_id, session_id):
+    """
+    Copy the converted nii image and its accompanying JSON sidecar from the temporary folder to the output BIDS folder.
+    For DWI images, also copy the bval and bvec files.
+
+    :param contrast: Contrast type, e.g., T2w, dwi
+    :param fname: Path to the converted nii image in the temporary folder
+    :param output_folder: temporary folder with the converted nii images
+    :param participant_id: participant ID, e.g., sub-001
+    :param session_id: session ID, e.g., ses-01
+    """
+    # First, create anat and dwi subfolders if they do not exist
+    if contrast == "dwi":
+        contrast_folder = "dwi"
+    else:
+        contrast_folder = "anat"
+    output_folder = os.path.join(output_folder, contrast_folder)
+    os.makedirs(output_folder, exist_ok=True)
+
+    # Second, move the images and JSON sidecars to the respective folders
+    fname_output = os.path.join(output_folder, f"{participant_id}_{session_id}_{contrast}.nii.gz")
+    print(f"Info: Copying {fname} to {fname_output}")
+    shutil.copy(fname, fname_output)
+    shutil.copy(fname.replace('.nii.gz', '.json'), fname_output.replace('.nii.gz', '.json'))
+    # For DWI, we also need to copy the bval and bvec files
+    if contrast == "dwi":
+        shutil.copy(fname.replace('.nii.gz', '.bval'), fname_output.replace('.nii.gz', '.bval'))
+        shutil.copy(fname.replace('.nii.gz', '.bvec'), fname_output.replace('.nii.gz', '.bvec'))
+
+
+def main():
+    """
+    Main function
+    """
+    args = get_parser()
+
+    dicom_folder = os.path.expanduser(args.dicom_folder)
+    bids_folder = os.path.expanduser(args.bids_folder)
+    participant_id = args.participant
+    session_id = args.session
+    contrasts = args.contrasts
+
+    print(100*"-")
+    print(f'Dicom folder: {dicom_folder}')
+    print(f'BIDS folder: {bids_folder}')
+    print(f'Participant ID: {participant_id}')
+    print(f'Session ID: {session_id}')
+    print(100*"-")
+
+    # Check if the folder with DICOMs exists
+    if not os.path.isdir(dicom_folder):
+        print(f"Error: Provided folder with DICOM images does not exist: {dicom_folder}")
+        exit(1)
+
+    # Check whether the BIDS folder already exists, if so, warn the user and exit
+    output_folder = os.path.join(bids_folder, participant_id, session_id)
+    if os.path.isdir(output_folder):
+        print(f"Error: BIDS folder for the provided participant and session already exists: {output_folder}"
+              f"\nPlease remove the existing BIDS folder and rerun the script.")
+        exit(1)
```

Comment:
maybe we want to asks if they want to overwrite it?

Reply:
Good point! I was considering this option. Then I decided on this workaround because if we overwrite from the script, we will lose the folder permanently. On the other hand, if the user deletes the folder manually (probably from Finder), the folder will be kept in the Trash folder if they need to recover it